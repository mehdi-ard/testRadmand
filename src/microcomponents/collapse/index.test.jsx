import React from 'react';
import renderer from 'react-test-renderer';
import Collapse from '.';

class AppCollapse extends React.Component {
  state = {
    data: [
      {
        title: 'A',
        arrow: true,
        text:
          'Lorem, ipsum dolor sit amet consectetur adipisicing elit. Saepe cum illum commodi autem quo non aliquid reprehenderit, odit perferendis impedit explicabo nemo? Pariatur sed iusto exercitationem qui ab fuga atque.'
      },
      {
        title: 'B',
        arrow: true,
        text:
          'Lorem, ipsum dolor sit amet consectetur adipisicing elit. Saepe cum illum commodi autem quo non aliquid reprehenderit, odit perferendis impedit explicabo nemo? Pariatur sed iusto exercitationem qui ab fuga atque.'
      },
      {
        title: 'C',
        arrow: false,
        text:
          'Lorem, ipsum dolor sit amet consectetur adipisicing elit. Saepe cum illum commodi autem quo non aliquid reprehenderit, odit perferendis impedit explicabo nemo? Pariatur sed iusto exercitationem qui ab fuga atque.'
      }
    ]
  };

  render() {
    const { data } = this.state;
    return <Collapse data={data} {...this.props} />;
  }
}

it('renders default', () => {
  expect(renderer.create(<AppCollapse />).toJSON()).toMatchSnapshot();
});

it('renders bordered', () => {
  expect(
    renderer.create(<AppCollapse bordered={false} />).toJSON()
  ).toMatchSnapshot();
});

it('renders expandIconPosition', () => {
  expect(
    renderer.create(<AppCollapse expandIconPosition="right" />).toJSON()
  ).toMatchSnapshot();
});

it('renders AppCollapse', () => {
  expect(renderer.create(<AppCollapse accordion />).toJSON()).toMatchSnapshot();
});
