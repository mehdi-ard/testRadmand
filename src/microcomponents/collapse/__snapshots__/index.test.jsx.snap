// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AppCollapse 1`] = `
<div
  className="ant-collapse ant-collapse-icon-position-left"
  role="tablist"
>
  <div
    className="ant-collapse-item"
  >
    <div
      aria-expanded={false}
      className="ant-collapse-header"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="tab"
      tabIndex={0}
    >
      <span
        aria-label="right"
        className="anticon anticon-right ant-collapse-arrow"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="right"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
          />
        </svg>
      </span>
      A
    </div>
  </div>
  <div
    className="ant-collapse-item"
  >
    <div
      aria-expanded={false}
      className="ant-collapse-header"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="tab"
      tabIndex={0}
    >
      <span
        aria-label="right"
        className="anticon anticon-right ant-collapse-arrow"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="right"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
          />
        </svg>
      </span>
      B
    </div>
  </div>
  <div
    className="ant-collapse-item ant-collapse-no-arrow"
  >
    <div
      aria-expanded={false}
      className="ant-collapse-header"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="tab"
      tabIndex={0}
    >
      C
    </div>
  </div>
</div>
`;

exports[`renders bordered 1`] = `
<div
  className="ant-collapse ant-collapse-borderless ant-collapse-icon-position-left"
  role={null}
>
  <div
    className="ant-collapse-item"
  >
    <div
      aria-expanded={false}
      className="ant-collapse-header"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        aria-label="right"
        className="anticon anticon-right ant-collapse-arrow"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="right"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
          />
        </svg>
      </span>
      A
    </div>
  </div>
  <div
    className="ant-collapse-item"
  >
    <div
      aria-expanded={false}
      className="ant-collapse-header"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        aria-label="right"
        className="anticon anticon-right ant-collapse-arrow"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="right"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
          />
        </svg>
      </span>
      B
    </div>
  </div>
  <div
    className="ant-collapse-item ant-collapse-no-arrow"
  >
    <div
      aria-expanded={false}
      className="ant-collapse-header"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      C
    </div>
  </div>
</div>
`;

exports[`renders default 1`] = `
<div
  className="ant-collapse ant-collapse-icon-position-left"
  role={null}
>
  <div
    className="ant-collapse-item"
  >
    <div
      aria-expanded={false}
      className="ant-collapse-header"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        aria-label="right"
        className="anticon anticon-right ant-collapse-arrow"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="right"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
          />
        </svg>
      </span>
      A
    </div>
  </div>
  <div
    className="ant-collapse-item"
  >
    <div
      aria-expanded={false}
      className="ant-collapse-header"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        aria-label="right"
        className="anticon anticon-right ant-collapse-arrow"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="right"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
          />
        </svg>
      </span>
      B
    </div>
  </div>
  <div
    className="ant-collapse-item ant-collapse-no-arrow"
  >
    <div
      aria-expanded={false}
      className="ant-collapse-header"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      C
    </div>
  </div>
</div>
`;

exports[`renders expandIconPosition 1`] = `
<div
  className="ant-collapse ant-collapse-icon-position-right"
  role={null}
>
  <div
    className="ant-collapse-item"
  >
    <div
      aria-expanded={false}
      className="ant-collapse-header"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        aria-label="right"
        className="anticon anticon-right ant-collapse-arrow"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="right"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
          />
        </svg>
      </span>
      A
    </div>
  </div>
  <div
    className="ant-collapse-item"
  >
    <div
      aria-expanded={false}
      className="ant-collapse-header"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        aria-label="right"
        className="anticon anticon-right ant-collapse-arrow"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="right"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
          />
        </svg>
      </span>
      B
    </div>
  </div>
  <div
    className="ant-collapse-item ant-collapse-no-arrow"
  >
    <div
      aria-expanded={false}
      className="ant-collapse-header"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      C
    </div>
  </div>
</div>
`;
